regex.c

```diff
@@ -164,79 +164,58 @@ void free_regex(void)
 	free_whitelist_domains();
 }
 
-static void read_whitelist_from_file(void)
+static void read_whitelist_from_database(void)
 {
-	FILE *fp;
-	char *buffer = NULL;
-	size_t size = 0;
-
-	// Get number of lines in the whitelist file
-	whitelist.count = countlines(files.whitelist);
+	// Get number of lines in the whitelist table
+	whitelist.count = gravityDB_count(WHITE_LIST);
 
 	if(whitelist.count < 0)
```

Comment:
This seems like it should check for equality as well, since it checks if there are any whitelist entries.

Reply:
The response could also have been `DB_FAILED` (which is defined as `-2`), but I agree that this should be separated out into another, more informative, message